baseplate/frameworks/pyramid/__init__.py

```diff
@@ -83,8 +139,43 @@ def baseplate_tween(request: Request) -> Response:
         else:
             if request.span:
                 request.span.set_tag("http.status_code", response.status_code)
+                content_length = response.content_length
                 response.app_iter = SpanFinishingAppIterWrapper(request.span, response.app_iter)
+                response.content_length = content_length
         finally:
+            http_endpoint = endpoint
+            http_method = request.method.lower()
+            http_response_code = ""
+
+            if sys.exc_info() == (None, None, None):
+                http_success = (
+                    getHTTPSuccessLabel(int(response.status_code)) if response else "false"
```

Comment:
Why the casting to an `int` here? is it not already an `int`?

Mainly wondering if we need any safety around here for when there is no response code (eg if the connection is closed, is this `None` or -1?)

Reply:
Yeah that cast is technically not required indeed.

This does make the assumption that this is not None. If the connection is unexpectedly closed it should be an exception which means that's the `else` code path.